datasets/imbalanced_dataset_sampler.py

```diff
@@ -1,4 +1,41 @@
-# TODO: Implement dataset sampler (issue #2)
+import itertools
+import typing
+import random
 
-class ImbalancedDatasetSampler:
-    pass
\ No newline at end of file
+import torch
+
+
+class ImbalancedDatasetSampler(torch.utils.data.sampler.Sampler):
+    """
+    Artificially samples classes from a multiclass dataset and create an imbalance according 
+    to the specified imbalance ratio (IR = n_first_class/n_last_class).
+    """
+    def __init__(self, dataset: torch.utils.data.dataset, imbalance_ratio: int, classes: typing.List[int]):
+        """Constructor
+
+        Args:
+            dataset (torch.utils.data.dataset): Original balanced dataset.
+            imbalance_ratio (int): Imbalance ratio (n_first_class/n_last_class).
+            classes (typing.List[int]): List of classes to sample from.
+        """
+        if imbalance_ratio != 1 and len(classes) == 1:
+            raise ValueError("Can't create an imbalance if only one classe is selected.")
+
+        class_idx = {}
```

Comment:
```suggestion
        class_idx = defaultdict(list)
```

Reply:
Slightly faster than doing the setdefault in loop
need to import `from collections import default dict`

Reply:
Whoa c’est donc ben nice les “suggested change” qu’on peut commit direct 🤯
Et oui c’est plus clean defaultdict, j’voulais faire qqch comme ça mais je savais pas comment 👌🏻